### ISSUE TITLE
Are Neural Rankers still Outperformed by Gradient Boosted Decision Trees?

### ISSUE THREAD

**Issue by hiroto0227** [hiroto0227/paper#3]
https://openreview.net/forum?id=Ut1vF_q_vC
2021年 Google

**Comment by hiroto0227** [comment]
### どんな論文か？
- Neural Networkではランキング学習(Learning To Rank(LTR))が、GBDTなどのTree系アルゴリズムよりも性能がよくないケースが多い。
- 原因として考えられるのは以下の三つである。
  - Neural Networkは入力にセンシティブなため、入力のスケールを合わせる必要がある。
  - ベンチマークに使用しているデータセットが比較的小規模なため、学習しきれていないか、overfitしている。
  - 一般の全結合層では情報が十分に伝わらないが、LTRのNNでは全結合層を使用していることが多い。
- それらに対応するためにNeural Networkの構造をランキング学習に最適化したところ、GBDTと同じかそれよりも良いスコアとなった。

**Comment by hiroto0227** [comment]
### 提案されたネットワーク構造

![スクリーンショット 2021-03-18 18 13 34](https://user-images.githubusercontent.com/22436847/111601423-ae022900-8815-11eb-876f-33b9b7f53f86.png)


### 各問題への対応
Neural Networkは入力にセンシティブなため、入力のスケールを合わせる必要がある。
-> 単純にlog1pをとってあげる。

ベンチマークに使用しているデータセットが比較的小規模なため、学習しきれていないか、overfitしている。
-> 入力に対してガウシアンノイズを加えて、Data Argumentを行う。

一般の全結合層では情報が十分に伝わらないが、LTRのNNでは全結合層を使用していることが多い。
-> multi-head-self-attentionを使用することで、複雑なネットワーク構造にしている。

**Comment by hiroto0227** [comment]
### Lossの選び方について
Appendix BでどのLossがLTRに最適化を議論している。
Pointwise Loss, Pairwise Loss, Listwise Loss の3つの種類から8つのロスを比較している。
Listwise Lossが高い傾向にあり、その中でもどのデータセットに対しても安定して高いスコアとなる & シンプルなSoftmax Cross Entropyを採用としている。

<img width="1002" alt="スクリーンショット 2021-03-18 18 19 00" src="https://user-images.githubusercontent.com/22436847/111602528-c3c41e00-8816-11eb-823c-1b22c8823efc.png">

**Comment by hiroto0227** [comment]
### 評価
- ベンチマークをGBDT系のSOTAである、lambdaMARTとしている。
- 提案手法のDASALCとlambdaMARTと比較すると、Web30Kでは買っているが、そのほかのデータセットでは少し劣っている。ただ、従来のNN手法よりは良い。
- Yahooのデータセットは公開時に入力値がすでにスケーリングされていたため、DASALCの精度の向上幅が小さかったと考えている。

<img width="991" alt="スクリーンショット 2021-03-18 18 22 28" src="https://user-images.githubusercontent.com/22436847/111602663-ed7d4500-8816-11eb-9c84-7e7ff2ab5d69.png">
